test_weather_alexander.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+import requests
+from requests import get
+
+
+# TODO: test other endpoints
+# TODO: add parametrization by the city
+
+
+new_set = set()
+
+
+def get_set_keys(iner_dict):
+    for k, v in iner_dict.items():
+        new_set.add(str(k))
+        if isinstance(v, dict):
+            get_set_keys(v)
+
+    return new_set
```

Comment:
not sure what the purpose of this function...
Respect for creating recursive function!
I see now why you defined new_set variable on the module level it will work here.
But it's really, really not a good practice.
If you use variables inside a standalone (independent) function - then define all variables inside that function.
Here is a solution to do that for this function:
```python
# you can pass same variable to the recursive calls
def get_set_keys(iner_dict, new_set=set()):
    for k, v in iner_dict.items():
        new_set.add(str(k))
        if isinstance(v, dict):
            get_set_keys(v, new_set)
    return new_set
```